tools/libSwiftSyntaxParser/libSwiftSyntaxParser.cpp

```diff
@@ -179,8 +196,68 @@ class CLibParseActions : public SyntaxParseActions {
     return {result.length, result.node};
   }
 };
+
+swiftparser_diagnostic_severity getSeverity(DiagnosticKind Kind) {
```

Comment:
Add `static`.

Reply:
Returning an enum value is also subject to ABI and API concerns if we decide to add a severity kind in the future/do portability work later.  Not sure how much that factors into the current design.

Reply:
What are your concerns ? AFAIK for enums in stable C APIs you only have to make sure to set specific integer values to the enum values and make sure to not change those integers in future versions. If you do add a new enum, bump the API version macro. This is what we have been doing with the libclang C API.

Reply:
So that handles the API concerns.

As for ABI, the underlying type of an enumeration is implementation-defined which means there’s a hypothetical world in which the addition of a case (say with a negative tag) causes the compiler to select a new layout and break everybody.

Reply:
This is not an issue because:
* This enum will just have a handful of values
* This library is intended to be used with a matching SwiftSyntax (their syntax definitions must match), you can't just mix-and-match them arbitrarily.